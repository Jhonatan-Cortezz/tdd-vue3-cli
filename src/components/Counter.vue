<script setup>
import { ref, computed } from 'vue'

/* props */
const props = defineProps({
  title: {
    type: String,
    required: false,
    default: 'None'
  },
  start: {
    type: Number,
    required: true
  }
})

const counter = ref(props.start)

/* methods */
const getSquareCounter = () => {
  console.log('getSquareCounter');
  return counter.value * counter.value
}

const increment = () => counter.value++

const decrement = () => counter.value--

/* computed */
const squareCounter = computed(() => {
  console.log('computedSquereCounter');
  return counter.value * counter.value
})
</script>
<template>
  <h2>{{ props.title }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <div>
    <button @click="increment">+1</button>
    <button @click="decrement">-1</button>
  </div>
</template>