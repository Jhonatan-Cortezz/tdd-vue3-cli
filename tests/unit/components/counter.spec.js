import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('counter component', () => {

  test('debe de ser igual al snaphot', () => {

    /* montamos en componente */
    const wrapper = shallowMount(Counter)

    /* indicamos q tome una captura y evalue si es igual al rederizado html*/
    expect(wrapper.html()).toMatchSnapshot()
  })

})