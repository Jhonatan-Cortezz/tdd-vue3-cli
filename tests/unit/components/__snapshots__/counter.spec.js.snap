// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter component debe de ser igual al snaphot 1`] = `
<h2>None</h2>
<p> <sup>2</sup> = NaN</p>
<div class="buttons-container"><button>+1</button><button>-1</button></div>
`;
